feat: Add custom root directory (#134)

j-tag opened the pull request:
As I mentioned in #134 , it would be great if we have the ability to set custom directory for blog so we can use the compiled version in another machine.

In this PR I added the functionality to set custom directory for blog.

roj1512 reviewed (commented):
Can you add a test for that?

J-TAG commented:
Yea for sure. I'll add test and update my PR

KorigamiK commented:
The name `customDirectory` should be renamed to something like `contentDirectory` to be more appropriate.

J-TAG commented:
> The name `customDirectory` should be renamed to something like `contentDirectory` to be more appropriate.

I'm adding the ability of setting base directory of blog in my PR. In my opinion `contentDirectory` should be the name of our property when we want to change the `posts` directory.

But in this PR the `posts` directory will still remain the same and will be like this: `customDirectory`  + `/posts`.
Correct me please if I'm wrong.

roj1512 commented:
How is `rootDirectory`?

J-TAG commented:
`rootDirectory` seems good to me. So I rename the settings property from `customDirectory` to `rootDirectory` in my PR?

roj1512 commented:
Yes, and you can refer it to as “custom root directory.”

J-TAG commented:
ok thank you. I'll update my PR

J-TAG commented:
I've changed the name to `rootDirectory`

roj1512 commented on blog_test.ts:
```diff
@@ -338,3 +338,28 @@ Deno.test("Plaintext response", async () => {
 });
+
+Deno.test(
+  "custom root directory for blog",
```
```suggestion
  "custom root directory",
```

roj1512 commented on testdata/customRootDir/posts/custom.md:
```diff
@@ -0,0 +1,9 @@
+---
+title: Custom post
+publish_date: 2022-03-20
+abstract: This is the first post in custom directory.
```
```suggestion
abstract: This is the first post with a custom root directory.
```

roj1512 commented on types.d.ts:
```diff
@@ -78,6 +78,8 @@ export interface BlogSettings {
   hostname?: string;
   /** Whether to display readtime or not */
   readtime?: boolean;
+  /** Custom root directory to looking for blog files and content */
```
```suggestion
  /** The root directory of the blog content */
```

J-TAG commented:
Thanks for the suggestions. I've applied them.

roj1512 reviewed (approved):
LGTM!

lino-levan commented:
@hashrock please review.

lino-levan reviewed (approved):
LGTM. Thanks for the test.

j-tag commented:
Thanks @hashrock for you review. My main reason for this PR is to be able to compile the blog in a machine, and host it on another. If static assets are not affected, it will not be useful. I'll try to check your point ASAP and update my PR if any changes needed.

j-tag commented:
@hashrock  I've checked your point and that was totally valid. Static files are not served with the previous version of my PR. So I've fixed my PR to support static files also.
The fact that I didn't catch this in previous version was that my browser was caching images and assets so I couldn't notice deno blog is not serving them. But after a hard refresh I could see that.

So basically now if you set the `rootDirectory` option, both posts and static asset files are gonna serve in that directory. 
Also I updated the tests to check static files with a custom `rootDirectory`.

Thank you for your review!

hashrock reviewed (approved):
LGTM! Thank you!
